frontend/src/lib/OutputArtifactLoader.ts

```diff
@@ -198,6 +217,275 @@ export class OutputArtifactLoader {
     };
   }
 
+  public static async getMlmdContext(argoPodName: string): Promise<Context> {
+    if (argoPodName.split('-').length < 3) {
+      throw new Error('kfpPodName has fewer than 3 parts');
+    }
+
+    // argoPodName has the general form "pipelineName-workflowId-executionId".
+    // All components of a pipeline within a single run will have the same
+    // "pipelineName-workflowId" prefix.
+    const pipelineName = argoPodName
+      .split('-')
+      .slice(0, -2)
+      .join('_');
+    const runID = argoPodName
+      .split('-')
+      .slice(0, -1)
+      .join('-');
+    const contextName = pipelineName + '.' + runID;
+
+    const request = new GetContextByTypeAndNameRequest();
+    request.setTypeName('run');
+    request.setContextName(contextName);
+    let res: GetContextByTypeAndNameResponse;
+    try {
+      res = await Api.getInstance().metadataStoreService.getContextByTypeAndName(request);
+    } catch (err) {
+      err.message = 'Failed to getContextsByTypeAndName: ' + err.message;
+      throw err;
+    }
+
+    const context = res && res.getContext();
+    if (!context) {
+      throw new Error("getContextByTypeAndName didn't have a context");
+    }
+    return context;
+  }
+
+  public static async getExecutionInContextWithPodName(
+    argoPodName: string,
+    context: Context,
+  ): Promise<Execution> {
+    const contextId = context.getId();
+    if (!contextId) {
+      throw new Error('Context must have an ID');
+    }
+
+    const request = new GetExecutionsByContextRequest();
+    request.setContextId(contextId);
+    let res: GetExecutionsByContextResponse;
+    try {
+      res = await Api.getInstance().metadataStoreService.getExecutionsByContext(request);
+    } catch (err) {
+      err.message = 'Failed to getExecutionsByContext: ' + err.message;
+      return err;
+    }
+
+    const executionList = (res && res.getExecutionsList()) || [];
+    const foundExecution = executionList.find(execution => {
+      const properties = execution.getPropertiesMap();
+      const executionPodName = properties.get('kfp_pod_name');
+      if (!executionPodName) {
+        return false;
+      }
+      const executionState = properties.get('state');
+      if (!executionState) {
+        return false;
+      }
+      return (
+        executionPodName.getStringValue() === argoPodName &&
+        executionState.getStringValue() === 'complete'
+      );
+    });
+    if (!foundExecution) {
+      logger.verbose("Couldn't find corresponding execution in context");
+      throw new Error("Couldn't find corresponding execution in context");
```

Comment:
This is expected error, so you should return undefined here and change return typing.